app/lib/lti_advantage/services/names_and_roles.rb

```diff
@@ -26,15 +26,33 @@ def valid?
       def list(query = nil)
         url = endpoint.dup
         url << "?#{query}" if query.present?
-        HTTParty.get(
-          url,
-          headers: headers(
-            {
-              "Content-Type" => "application/vnd.ims.lti-nrps.v2.membershipcontainer+json",
-            },
+
+        verify_received_learner_names(
+          HTTParty.get(
+            url,
+            headers: headers(
+              {
+                "Content-Type" => "application/vnd.ims.lti-nrps.v2.membershipcontainer+json",
+              },
+            ),
           ),
         )
       end
+
+      def verify_received_learner_names(names_and_roles_memberships)
+        if names_and_roles_memberships.present?
+          members = JSON.parse(names_and_roles_memberships.body)["members"]
+
+          if members.present? && members.all? { |member| member["name"].nil? }
+            raise(
+              GraphQL::ExecutionError,
```

Comment:
I think we should throw an LTI Advantage specific exception here and then catch it in the graphql code - in the query or mutation. Then we can throw the GraphQL specific error from there.